h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java

```diff
@@ -147,6 +152,25 @@ protected void setMetalearnerParameters(StackedEnsembleParameters params) {
                 params.initMetalearnerParams(_metalearnerAlgo);
                 params._metalearner_parameters._keep_cross_validation_models = buildSpec.build_control.keep_cross_validation_models;
                 params._metalearner_parameters._keep_cross_validation_predictions = buildSpec.build_control.keep_cross_validation_predictions;
+                setDistributionParameters(params._metalearner_parameters);
+            }
+
+            @Override
+            public boolean supportsDistribution(DistributionFamily distributionFamily) {
+                StackedEnsembleParameters params = prepareModelParameters();
+                setMetalearnerParameters(params);
+                try {
+                    params.setDistributionFamily(distributionFamily);
+                } catch (H2OIllegalArgumentException e) {
+                    return false;
+                }
+                // Just to get the correct response
+                setCommonModelBuilderParams(params);
+                ModelBuilder mb = ModelBuilder.make(params);
+                mb.init(false);
+                return Arrays.stream(new String[]{"_distribution", "_family"})
+                        .allMatch((field) ->
+                                mb.getMessagesByFieldAndSeverity(field, Log.ERRR).length == 0);
```

Comment:
if this is the only reason why you need this `supportsDistribution` method, then there must be a better alternative: this is too complex and redundant.

You already have `StackedEnsembleModel.setDistributionFamily()` forwarding to the metalearner. So standard validation logic should check everything correctly in `StackedEnsemble.init()`: nothing that looks specific to SEs in AutoML, it should be done for any standard SE.

Reply:
Good point, thank you! `supportsDistribution` is no longer used so I removed it completely.